samples/MvcSample.Web/HomeController.cs

```diff
@@ -90,6 +91,23 @@ public ActionResult SaveUser(User user)
         }
 
         /// <summary>
+        /// Action that shows multiple file upload.
+        /// </summary>
+        public async Task<ActionResult> PostFile(IEnumerable<IFormFile> file)
+        {
+            if (!ModelState.IsValid)
+            {
+                return View("MyView");
+            }
+
+            foreach (var f in file)
+            {
+                await f.SaveAsAsync("C:\\Temp\\test-" + f.ParseContentDisposition().Value);
```

Comment:
Or intrinsically save names: Path.GetTempFileName (is it available on .NET Core?)